package.json

```diff
@@ -69,7 +68,7 @@
     "lint": "pnpm lint:sol && pnpm lint:ts && pnpm prettier:check",
     "lint:sol": "solhint --max-warnings 0 \"contracts/**/*.sol\"",
     "lint:ts": "eslint --ignore-path ./.eslintignore --ext .js,.ts .",
-    "postinstall": "DOTENV_CONFIG_PATH=./.env.example pnpm typechain",
+    "postcompile": "pnpm typechain",
```

Comment:
We changed this from `postinstall` to `postcompile` because otherwise the `typechain` task would be executed after running `pnpm i`, but it's likely that the user hasn't set the configuration variables yet.